core/src/main/java/org/apache/calcite/rex/RexSimplify.java

```diff
@@ -1571,6 +1571,9 @@ RexNode simplifyAnd2ForUnknownAsFalse(List<RexNode> terms,
 
   private <C extends Comparable<C>> RexNode simplifyUsingPredicates(RexNode e,
       Class<C> clazz) {
+    if (predicates.pulledUpPredicates.isEmpty()) {
+      return e;
```

Comment:
What if we do not return early ? Is there any error occurs ?

Reply:
No, but it would bring some extra unnecessary overhead since all the following process still need to execute.